### PR TITLE
Adu update bash script

### PR DESCRIPTION
**Pull request by JeffMill**
Bug 32485748: Create bash script to generate import manifest (parity with AduUpdate.psm1)

bash version of the PowerShell script.

Bonus: Split README to PowerShell and bash versions.

Bonus: Fixed up some stuff in the PowerShell README

### CODE REVIEW

File: tools/AduCmdlets/README-bash.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+Bash script `create-adu-import-manifest.sh` can be used to create an *import manifest* to import an update into Azure Device Update.
+
+## Usage
+
+To create an import manifest:
+
+```bash
+# Create an update that will be compatible with devices from two different manufacturers.
+
+./create-adu-import-manifest.sh -p 'Microsoft' -n 'Toaster' -v '2.0' -t 'microsoft/swupdate:1' -i'5' -c Fabrikam,Toaster -c Contoso,Toaster ./file1.json ./file2.zip
```
Comment (JeffMill) [inline tools/AduCmdlets/README-bash.md:14]: >-i'5' [](start = 95, length = 5)

I should make this `i '5'` to be consistent.

File: tools/AduCmdlets/create-adu-import-manifest.sh [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+for file in "${UPDATE_FILES[@]}"; do
+    if [ ! -f "$file" ]; then
+        write_error "File '$file' not found."
+        exit 1
+    fi
+done
+
+# Match Powershell's .ToString('o') format, e.g. "2021-04-14T18:14:35.5816196Z"
+CREATED_DATETIME=$(date --utc "+%Y-%m-%dT%H:%M:%S.%NZ")
+
```
Comment (JeffMill) [inline tools/AduCmdlets/create-adu-import-manifest.sh:226]: checking with Leo, but `date --utc --iso-8601=seconds` might suffice here if ns aren't significant, and .NET DateTime is being used to parse.

File: tools/AduCmdlets/create-adu-import-manifest.sh [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+# Write out JSON.
+# Using "jq" would be better, but trying to eliminate script dependencies.
```
Comment (JeffMill) [inline tools/AduCmdlets/create-adu-import-manifest.sh:228]: >eliminate [](start = 44, length = 9)

reduce (openssl is a dependency)

**Review by devlie — CHANGES_REQUESTED** [review]

File: tools/AduCmdlets/README-bash.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
+# create-adu-import-manifest.sh Bash Script
+
+## Overview
+
+Bash script `create-adu-import-manifest.sh` can be used to create an *import manifest* to import an update into Azure Device Update.
```
[the remark this answers is not in the document]
Comment (JeffMill) [reply tools/AduCmdlets/README-bash.md:5]: yeah, but the powershell script and the folder name still have "adu".  I suppose I can change the MD files for now.

---
In reply to: [613560343](https://github.com/Azure/iot-hub-device-update/pull/41#discussion_r613560343) [](ancestors = 613560343)

File: tools/AduCmdlets/create-adu-import-manifest.sh [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+ENABLE_VERBOSE=0
+
+# This is the version of the ADU update manifest schema being used.
+MANIFEST_VERSION='2.0'
+
+print_usage() {
+    printf "Usage: create-adu-import-manifest [OPTION]... [FILE]...
+Create an import manifest to import an update into Device Update.
+
+-p NAME         Update provider name
```
[the remark this answers is not in the document]
Comment (JeffMill) [reply tools/AduCmdlets/create-adu-import-manifest.sh:14]: so remove "name" here?

---
In reply to: [613586257](https://github.com/Azure/iot-hub-device-update/pull/41#discussion_r613586257) [](ancestors = 613586257)

[the remark this answers is not in the document]
Comment (devlie) [reply tools/AduCmdlets/create-adu-import-manifest.sh]: Yeah, 
`-p PROVIDER         Update provider`

**Review by dpokluda — APPROVED** [review]

**Pull request merged**